packages/stark-ui/src/modules/transform-input-directive/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,385 @@
+	const inputEvent: Event = document.createEvent("Event");
+	inputEvent.initEvent("input", true, true);
+
+	element.value = value;
+	element.dispatchEvent(inputEvent);
+}
+
+describe("TransformInputDirective", () => {
+	describe("with ngModel", () => {
+		@Component({
+			selector: "test-component",
+			template: "<input [(ngModel)]='value' [starkTransformInput]='starkTransformInputValue'/>"
+		})
+		class TestComponent {
+			public starkTransformInputValue: StarkInputTransformationType = () => {
+				/*noop*/
+			};
+			public value: string = "";
+		}
+
+		let fixture: ComponentFixture<TestComponent>;
+		let component: TestComponent;
+		let htmlInputElement: HTMLInputElement;
+
+		beforeEach(() => {
+			TestBed.configureTestingModule({
+				declarations: [StarkTransformInputDirective, TestComponent],
+				imports: [FormsModule]
+			});
+
+			fixture = TestBed.createComponent(TestComponent);
+			component = fixture.componentInstance;
+			htmlInputElement = fixture.nativeElement.querySelector("input");
+
+			// trigger initial data binding
+			fixture.detectChanges();
+
+			expect(component.value).toBe("", "field 'value' should start as empty string ");
+		});
+
+		it("should set value to uppercase", () => {
+			const input: string = "upper";
+			const expected: string = "UPPER";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = "uppercase";
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(component.value).toBe(expected);
+		});
+
+		it("should set value to lowercase", () => {
+			const input: string = "LOWER";
+			const expected: string = "lower";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = "lowercase";
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(component.value).toBe(expected);
+		});
+
+		it("should replace dirty word", () => {
+			const input: string = "fudge you!";
+			const expected: string = "***** you!";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = (v: string) =>
+				v.replace("fudge", (match: string) =>
+					match
+						.split("")
+						.map(() => "*")
+						.join("")
+				);
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(component.value).toBe(expected);
+		});
+	});
+
+	describe("with formControl", () => {
+		@Component({
+			selector: "test-component",
+			template: "<input [formControl]='formControl' [starkTransformInput]='starkTransformInputValue'/>"
+		})
+		class TestComponent {
+			public starkTransformInputValue: StarkInputTransformationType = () => {
+				/*noop*/
+			};
+			public formControl: FormControl = new FormControl("");
+
+			public constructor() {
+				this.formControl.valueChanges.subscribe(this.changeObserver);
```

Comment:
Is this necessary? I don't think so.... so you could also remove the `changeObserver()` method